buildSrc/src/main/java/com/google/firebase/gradle/plugins/DackkaPlugin.kt

```diff
@@ -0,0 +1,162 @@
+        javadocConfig.attributes.attribute(
+            BuildTypeAttr.ATTRIBUTE,
+            project.objects.named(BuildTypeAttr::class.java, "release")
+        )
+    }
+
+    private fun registerGenerateDackkaDocumentationTask(project: Project) =
+        project.tasks.register<GenerateDocumentationTask>("generateDackkaDocumentation") {
+            project.extensions.getByType<LibraryExtension>().libraryVariants.all {
+                if (name == "release") {
+                    mustRunAfter("createFullJarRelease")
+                    dependsOn("createFullJarRelease")
+
+                    val classpath = project.provider {
+                        runtimeConfiguration.getJars() + project.javadocConfig.getJars() + project.bootClasspath
+                    }
+
+                    val sourcesForJava = sourceSets.flatMap {
+                        it.javaDirectories.map { it.absoluteFile } + projectSpecificSources(project)
+                    }
+
+                    // this will become useful with the agp upgrade, as they're separate in 7.x+
+                    val sourcesForKotlin = emptyList<File>()
+                    val excludedFiles = emptyList<File>() + projectSpecificSuppressedFiles(project)
+
+                    dependencies.set(classpath)
+                    javaSources.set(sourcesForJava)
+                    kotlinSources.set(sourcesForKotlin)
+                    suppressedFiles.set(excludedFiles)
+
+                    applyCommonConfigurations()
+                }
+            }
+        }
+
+    // Remove when fixed: b/243534168
+    private fun projectSpecificSources(project: Project) =
+        when (project.name) {
+            "firebase-common" -> {
+                project.project(":firebase-firestore").files("src/main/java/com/google/firebase").toList()
+            }
+            else -> emptyList()
+        }
+
+    // Remove when fixed: b/243534168
+    private fun projectSpecificSuppressedFiles(project: Project): List<File> =
+        when (project.name) {
+            "firebase-common" -> {
+                val firestoreProject = project.project(":firebase-firestore")
+                firestoreProject.files("src/main/java/com/google/firebase/firestore").toList()
+            }
+            "firebase-firestore" -> {
+                project.files("src/main/java/com/google/firebase/Timestamp.java").toList()
+            }
+            else -> emptyList()
+        }
+
+    private fun GenerateDocumentationTask.applyCommonConfigurations() {
+        val dackkaFile = project.provider { project.dackkaConfig.singleFile }
+        val dackkaOutputDirectory = File(project.buildDir, "dackkaDocumentation")
+
+        dackkaJarFile.set(dackkaFile)
+        outputDirectory.set(dackkaOutputDirectory)
+    }
+
+    private fun registerFiresiteTransformTask(project: Project) =
+        project.tasks.register<FiresiteTransformTask>("firesiteTransform") {
+            mustRunAfter("generateDackkaDocumentation")
```

Comment:
is must run after the right thing to do as opposed to connecting the task's output to this task's input?

i.e. 
```kotlin
val dackka = project.tasks.register<GenerateDocumentationTask>("generateDackkaDocumentation") {
  // ...
}
project.tasks.register<FiresiteTransformTask>("firesiteTransform") {
  dackkaFiles.set(dackka.flatMap { it. outputDirectory })
}
```

context: https://docs.gradle.org/current/userguide/lazy_configuration.html#working_with_task_dependencies_in_lazy_properties

Reply:
Thats a really good idea, I didn't even think about that. That also avoids hardcoded strings. +1